users.go

```diff
@@ -57,33 +57,31 @@ type UpdateUserResponse struct {
 }
 
 func (c *Client) UpdateUser(ctx context.Context, params UpdateUserParams) (*User, error) {
-	var user *User
-
-	if params.Password != "" || params.Email != "" {
-		req := &http.APIRequest{
-			Method:   "PATCH",
-			Endpoint: "/account/profile",
-			Params: map[string]interface{}{
-				"user": params,
-			},
-			Expected: http.Statuses{200},
-		}
-		res, err := c.AuthAPI().Do(ctx, req)
-		if err != nil {
-			return nil, errgo.Notef(err, "fail to execute the query to update the user")
-		}
-		defer res.Body.Close()
-
-		var u UpdateUserResponse
-		err = json.NewDecoder(res.Body).Decode(&u)
-		if err != nil {
-			return nil, errgo.Notef(err, "fail to decode response of the query to update the user")
-		}
-
-		user = u.User
+	if params.Password == "" || params.Email == "" {
```

Comment:
question: I think the exact opposite of the current condition is:
```suggestion
	if params.Password == "" && params.Email == "" {
```

Isn't it? I agree that it may not be the ideal condition but I don't really want to change the behaviour in a refactoring PR :)

Reply:
yes you are right, however i guess AuthApi would always fail if one of the param is missing. But if you want to keep the same behaviour, i will commit your suggestion 👍🏽

Reply:
Yes indeed, the backend is already checking this so it's not that much important. It's really just a matter of keeping the existing behaviour.

Reply:
Thanks for the clarification !